api/src/main/java/org/apache/iceberg/exceptions/FileHandlingException.java

```diff
@@ -0,0 +1,29 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.exceptions;
+
+import com.google.errorprone.annotations.FormatMethod;
+
+/** Exception raised when unexpected behaviour happens in file operations. */
+public class FileHandlingException extends RuntimeException {
```

Comment:
I'm confused by the end goal, we're getting rid of the `RuntimeIOException`, but are creating a new exception, namely `FileHandlingException`. Shouldn't we just replace the `FileHandlingException` with `UncheckedIOException`? I think that was the goal looking at https://github.com/apache/iceberg/pull/1154

Reply:
`FileHandlingException` is for `RuntimeIOException` without an underlaying `IOException` cause.
For this kind, throwing `UncheckedIOException`  when there is no `IOException` underneath is very odd. Hence throwing a `RuntimeException` (Since Iceberg wraps `RuntimeException`  to be a known exception), by introducing new `FileHandlingException`.

Ryan only suggested to throw different exception for this kind (https://github.com/apache/iceberg/pull/4776#discussion_r875038397) long back instead of  `UncheckedIOException`

https://github.com/apache/iceberg/pull/4776#discussion_r875038397

Reply:
Regarding end goal. 

`RuntimeIOException` has two kinds occurrence in code.
a) One is with the cause of `IOException`.  (102 places). --> It is replaced blindly by `UncheckedIOException`  -> PR #4776

b) One without any exception cause. (9 places)  -> PR #6887
Explained above.